features/support/utils_gen.rb

```diff
@@ -0,0 +1,222 @@
+    screenshot(screenshot_path)
+  end
+  embed(screenshot_path, "image/png", "SCREENSHOT")
+end
+
+
+def embed_screenshot scenario
+  name = scenario.name.gsub(" ", "_").downcase
+  screenshot_path = File.dirname(__FILE__) + '/media/' + ENV['platform'] + '/'
+  screenshot_embed(options = {:prefix => screenshot_path, :name => name})
+end
+
+def wait_until_loader_disappear
+  while query("* id:'wait_spinner'").size != 0
+    sleep 0.5
+  end
+end
+
+def wait_until_text_disappear(text)
+  while check_if_text_exists(text)
+    sleep 0.5
+  end
+end
+
+def wait_until_text_visible(text)
+  wait_until_element_ready "* {text CONTAINS[C] '#{text}'}"
+end
+
+def wait_until_element_ready(qr)
+  10.times do
+    if query(qr).size > 0
+      break
+    else
+      sleep 1
+      puts "- Element is not ready yet ...."
+    end
+  end
+end
+
+def wait_until_one_element_ready(*args)
+  # this will wait until one of the element visible in the given query list
+  # wait_until_one_element_ready("* text:'Stores'", "* text:'Places'")
+
+  element = :not_exist
+
+  10.times do
+    args.each do |qr|
+      if query(qr).size > 0
+        element = :exist
+        break
+      end
+    end
+
+    if element == :exist
+      break
+    else
+      sleep 1
+      puts "- Element is not ready yet ...."
+    end
+  end
+end
+
+
+def check_obj_if_inside_the_container(container, obj)
+  wait_for_text("Today", timeout: 10)
+
+  # container: uniquery, contaioner object
+  # obj: uniquery, will be check if inside the container
+  #
+  # x1                 y1
+  #
+  #    obj
+  #
+  # x2                 y2
+
+  container = query(container)[0]
+  obj = query(obj)[0]
+
+  # boundaries = [x1, y1, x2, y2]
+  x1, y1, x2, y2 = [container["rect"]["x"].to_i, container["rect"]["y"].to_i, container["rect"]["x"].to_i + container["rect"]["width"].to_i, container["rect"]["y"].to_i + container["rect"]["height"].to_i]
+  obj_x, obj_y = obj["rect"]["center_x"].to_i, obj["rect"]["center_y"].to_i
+
+  # puts x1, y1, x2, y2
+  # puts obj_x, obj_y
+
+  (obj_x.between?(x1, x2) and obj_y.between?(y1, y2)) ? true : false
+end
+
+
+def reinstall_ios_app
+  %x{
+		if [[ $(brew list|grep "ideviceinstaller"|wc -l) -eq 0 ]]; then
+			echo -e " - Not found 'ideviceinstaller' for ios app installer. \n - installing ideviceinstaller"
+			brew uninstall ideviceinstaller
+			brew install --HEAD libimobiledevice
+			brew link --overwrite libimobiledevice
+			brew install --HEAD  ideviceinstaller
+			brew link --overwrite ideviceinstaller
+			# sudo chmod -R 777 /var/db/lockdown/
+		else
+			echo -e "\n - ideviceinstaller has been already installed."
+		fi
+	}
+
+  system("ideviceinstaller -U '#{ENV["BUNDLE_ID"]}'")
+
+  puts "- App removed, re-installing the app"
+
+  system("ideviceinstaller -i '#{ENV["APP_BUNDLE_PATH"]}'")
+end
+
+def ios_app_installed?
+  system("ideviceinstaller -l | grep #{ENV['BUNDLE_ID']}") == true
+end
+
+def check_if_element_exists query
+  query(query).size != 0 ? true : false
```

Comment:
change to `element_exist(query)`
```suggestion
  element_exist(query)
```